src/pygeoprocessing/geoprocessing.py

```diff
@@ -1798,19 +1798,24 @@ def reclassify_raster(
     """Reclassify pixel values in a raster.
 
     A function to reclassify values in raster to any output type. By default
-    the values except for nodata must be in ``value_map``.
+    the values except for nodata must be in ``value_map``. If the base
+    raster nodata value is in ``value_map``, then nodata values are
+    reclassified to the mapped value. This would essentially leave no nodata
+    holes in the target raster.
 
     Args:
         base_raster_path_band (tuple): a tuple including file path to a raster
             and the band index to operate over. ex: (path, band_index)
         value_map (dictionary): a dictionary of values of
             {source_value: dest_value, ...} where source_value's type is the
             same as the values in ``base_raster_path`` at band ``band_index``.
-            Must contain at least one value.
+            Cannot be empty and must contain a mapping for all raster values
+            if ``values_required=True``. If nodata is mapped, nodata will be
+            reclassified, otherwise nodata will be set to ``target_nodata``.
         target_raster_path (string): target raster output path; overwritten if
             it exists
         target_datatype (gdal type): the numerical type for the target raster
-        target_nodata (numerical type): the nodata value for the target raster
+        target_nodata (numerical type): the nodata value for the target raster.
```

Comment:
```suggestion
        target_nodata (numerical type): the nodata value for the target raster.
            Must be the same type as target_datatype. All nodata pixels in the 
            base raster will be reclassified to this value, and the target 
            raster's nodata value will be set to this value.
```

Reply:
Looks like your multi-line suggestion worked, Emily!

Reply:
I'm not sure if it will overwrite line 1819 also, or just add in between them

Reply:
I didn't commit your exact suggestion but put pieces of it in here: https://github.com/natcap/pygeoprocessing/commit/31b9c7164353426e15d7c43d801d3e5823038e57

I basically just elaborated a little onto yours and left the last part: `and the target raster's nodata value will be set to this value.` out since it's already mentioned in the first sentence.

Let me know what you think!

Reply:
I like your version! These things are often tricky to describe neatly.